From ce8e497bbdeed32a8357b7ac475b22df2654e1a6 Mon Sep 17 00:00:00 2001
From: Rajesh Bhagat <rajesh.bhagat@nxp.com>
Date: Fri, 1 Sep 2017 09:09:37 +0530
Subject: [PATCH 137/155] ls1088a: Correct the channel number for LTC3882
 voltage regulator

Corrects the channel number passed in i2c write operation for
LTC3882 voltage regulator.

Signed-off-by: Ashish Kumar <Ashish.Kumar@nxp.com>
Signed-off-by: Rajesh Bhagat <rajesh.bhagat@nxp.com>
---
 board/freescale/common/vid.c      | 3 ++-
 board/freescale/ls1088a/ls1088a.c | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/board/freescale/common/vid.c b/board/freescale/common/vid.c
index dcff1d1953..24bdb66d6f 100644
--- a/board/freescale/common/vid.c
+++ b/board/freescale/common/vid.c
@@ -176,10 +176,11 @@ static int read_voltage_from_IR(int i2caddress)
 static int read_voltage_from_LTC(int i2caddress)
 {
 	int  ret, vcode = 0;
+	u8 chan = PWM_CHANNEL0;
 
 	/* select the PAGE 0 using PMBus commands PAGE for VDD*/
 	ret = i2c_write(I2C_VOL_MONITOR_ADDR,
-			PMBUS_CMD_PAGE, 1, PWM_CHANNEL0, 1);
+			PMBUS_CMD_PAGE, 1, &chan, 1);
 	if (ret) {
 		printf("VID: failed to select VDD Page 0\n");
 		return ret;
diff --git a/board/freescale/ls1088a/ls1088a.c b/board/freescale/ls1088a/ls1088a.c
index fbdc24589e..a2b9b687d1 100644
--- a/board/freescale/ls1088a/ls1088a.c
+++ b/board/freescale/ls1088a/ls1088a.c
@@ -316,10 +316,11 @@ int i2c_multiplexer_select_vid_channel(u8 channel)
 int get_serdes_volt(void)
 {
 	int  ret, vcode = 0;
+	u8 chan = PWM_CHANNEL0;
 
 	/* Select the PAGE 0 using PMBus commands PAGE for VDD */
 	ret = i2c_write(I2C_SVDD_MONITOR_ADDR,
-			PMBUS_CMD_PAGE, 1, PWM_CHANNEL0, 1);
+			PMBUS_CMD_PAGE, 1, &chan, 1);
 	if (ret) {
 		printf("VID: failed to select VDD Page 0\n");
 		return ret;
-- 
2.14.1

